controllers/storageclass/storageclass_controller.go

```diff
@@ -118,52 +118,69 @@ func (r *StorageClassWatcher) Reconcile(_ context.Context, request reconcile.Req
 		return result, err
 	}
 
-	flashSystemCluster, fscErr := r.getFlashSystemClusterByStorageClass(sc)
-	if fscErr == nil {
-		fscName := flashSystemCluster.GetName()
-		r.Log.Info("FlashSystemCluster found", "FlashSystemCluster", fscName)
-
-		// Check GetDeletionTimestamp to determine if the object is under deletion
-		if !sc.GetDeletionTimestamp().IsZero() {
-			r.Log.Info("object is terminated")
-			err = r.removeStorageClassFromConfigMap(*configMap, fscName, request.Name)
-			if err != nil {
-				return result, err
-			}
-		} else {
-			poolName, ok := sc.Parameters[util.CsiIBMBlockScPool]
-			if ok {
-				if err = r.addStorageClassToConfigMap(*configMap, fscName, request.Name, poolName); err != nil {
+	_, isTopology := sc.Parameters[util.TopologyStorageClassByMgmtId]
+	if isTopology {
+		r.Log.Info("StorageClass is a topology aware StorageClass", "isTopology", isTopology)
+	}
+	fscToPoolsMap, fscErr := r.getFlashSystemClusterByStorageClass(sc, isTopology)
+	for fsc, poolName := range fscToPoolsMap {
+		if fscErr == nil {
```

Comment:
`fscErr` is shared between all flashsystemclusters, no need to verify it in every loop iteration